nixos/modules/services/networking/netbird.nix

```diff
@@ -7,162 +7,596 @@
+                type = nullOr str;
+                default = null;
+                example = "127.0.0.123";
                 description = ''
-                  Port for the ${name} netbird interface.
+                  An explicit address that Netbird will serve `*.netbird.cloud.` (usually) entries on.
+
+                  Netbird serves DNS on it's own (dynamic) client address by default.
                 '';
               };
 
+              dns-resolver.port = mkOption {
+                type = port;
+                default = 53;
+                description = ''
+                  A port to serve DNS entries on when `dns-resolver.address` is enabled.
+                '';
+              };
+
+              interface = mkOption {
+                type = str;
+                default = "nb-${client.name}";
+                description = ''
+                  Name of the network interface managed by this client.
+                '';
+                apply =
+                  iface:
+                  lib.throwIfNot (
+                    builtins.stringLength iface <= 15
+                  ) "Network interface name must be 15 characters or less" iface;
+              };
+
               environment = mkOption {
                 type = attrsOf str;
                 defaultText = literalExpression ''
                   {
-                    NB_CONFIG = "/var/lib/''${stateDir}/config.json";
-                    NB_LOG_FILE = "console";
-                    NB_WIREGUARD_PORT = builtins.toString port;
-                    NB_INTERFACE_NAME = name;
-                    NB_DAMEON_ADDR = "/var/run/''${stateDir}"
+                    NB_CONFIG = "/var/lib/netbird-''${client.name}/config.json";
+                    NB_DAEMON_ADDR = "unix:///var/run/netbird-''${client.name}/sock";
+                    NB_INTERFACE_NAME = client.interface;
+                    NB_LOG_FILE = mkOptionDefault "console";
+                    NB_LOG_LEVEL = client.logLevel;
+                    NB_SERVICE = "netbird-''${client.name}";
+                    NB_WIREGUARD_PORT = toString client.port;
+                  } // optionalAttrs (client.dns-resolver.address != null) {
+                    NB_DNS_RESOLVER_ADDRESS = "''${client.dns-resolver.address}:''${builtins.toString client.dns-resolver.port}";
                   }
                 '';
                 description = ''
                   Environment for the netbird service, used to pass configuration options.
                 '';
               };
 
-              stateDir = mkOption {
-                type = str;
-                default = "netbird-${name}";
+              autoStart = mkOption {
+                type = bool;
+                default = true;
                 description = ''
-                  Directory storing the netbird configuration.
+                  Start the service with the system.
+
+                  As of 2024-02-13 it is not possible to start a Netbird client daemon without immediately
```

Comment:
There is an `disable-auto-connect` flag but it seems to only apply to `netbird up`, the link however seems to have run stale as there is no mention of such a feature being planned as far as I can see.

Reply:
gotta look into it, I'm using some form of this in my configs successfully